### PR TITLE
Add basic test config and run tests via CI

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

* Adds [`ts-jest`](https://kulshekhar.github.io/ts-jest/), associated dependencies and config
    * I'm not that familiar with testing frameworks for TypeScript, but we are using this for other Developer Experience projects (e.g. `cdk`), so it seemed like a sensible choice to me
* Runs tests as part of CI
    * I've also updated TeamCity to report commit status back to GitHub
* Adds a couple of simple tests to prove that this works
    * I needed to refactor slightly to allow this code to be tested without [interacting with environment variables](https://github.com/guardian/cloudwatch-logs-management/blob/60f8bddb14c3ec27e327f4a7123300a0569ba801/src/shipLogEntries.ts#L12) which are unrelated to this function

## How to test

Tests are now being run [via CI](https://teamcity.gutools.co.uk/buildConfiguration/Tools_CloudwatchLogsManagement/633481?buildTab=log&focusLine=3&linesState=39). You can also run `npm run test` locally.

## How can we measure success?

It's possible to unit test functionality in the future. I'll attempt to add some more meaningful tests in a subsequent PR.

## Have we considered potential risks?

I can't think of any...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,12 +8,16 @@
...
+    "test": "jest"
   },
   "devDependencies": {
     "@types/aws-lambda": "^8.10.20",
+    "@types/jest": "^26.0.20",
     "@types/node": "12.12.7",
-    "typescript": "^3.3.3333"
+    "jest": "^26.6.3",
+    "ts-jest": "^26.5.3",
+    "typescript": "^3.9.9"
```
Comment (akash1810) [inline package.json:20]: Any reason to not move to v4? We can do this in a following PR too.

File: scripts/ci.sh [hunk 2]
```diff
@@ -30,8 +30,9 @@ fi
 nvm install
 nvm use
 
-# install deps and build app
+# install deps, run tests and build app
 npm install
```
Comment (akash1810) [inline scripts/ci.sh:34]: This should really be `npm ci` - https://docs.npmjs.com/cli/v7/commands/npm-ci.

**Review by akash1810 — APPROVED** [review]
Added a comment against the CI script, else LGTM!

File: package.json (hunk 1 above)
Comment (jacobwinch) [reply package.json:20]: This wasn't actually a deliberate change - I guess it must have been bumped automatically when I was installing new dependencies 🤔 

> We can do this in a following PR too.

Yeah, a major version bump probably feels worthy of its own PR, so I will make that change separately?

Comment (jacobwinch) [reply package.json:20]: https://github.com/guardian/cloudwatch-logs-management/pull/45

**Pull request merged**